src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1089,6 +1092,13 @@ private void EnforceDoesNotReturn(SyntaxNode? syntaxOpt)
             Analyze(compilation, method, node, diagnostics, useConstructorExitWarnings, initialNullableState, getFinalNullableState, out finalNullableState);
         }
 
+#if DEBUG
+        private static bool ShouldRunAnalysisInDebug(CSharpCompilation compilation)
```

Comment:
Think the method name is a bit off here to what it actually does. Think something closer to `ShouldRunAnalysisAndIgnoreResults` is closer.